### ISSUE TITLE
hotbar has unknown or dosent respect hud api z-index

### ISSUE THREAD

**Issue by sirrobzeroone** [minetest/minetest#11565]
##### Minetest version
**5.4**

##### Summary
Probably a lack of knowledge for myself however when adding a new hud image using hud_add, no matter what the z-index is set too I cant get the added image to sit under "gui_hotbar.png". I tried seeing if i could set hotbars z-index but im not sure it exists as far as hud api is concerned (part hard coded?).

api appears to indicate hotbar has a z-index of 0?
https://github.com/minetest/minetest/blob/0c1e9603db32b4281974fdd8b8e3b505148be47e/doc/lua_api.txt#L1398

##### Steps to reproduce
```
minetest.register_on_joinplayer(function(player)
	-- Set hotbar test texture	
  player:hud_add({
	  hud_elem_type = "image",
	  position = {x=0.5,y=0.95},
	  scale = { x=1, y=1 },
	  text = "some_image.png",
	  offset = {x=0, y=0},
	  z_index = -300
  })				
end)
```
expected outcome: some_image.png will be under hotbar, always draws above. I get the same behaviour with crosshairs.

**Comment by sfan5** [comment]
The documentation documents the ideal state, not what the engine actually does. Of course this doesn't make any sense and I pointed this out https://github.com/minetest/minetest/pull/9701#issuecomment-615477898 but nobody seemed to care.

**Comment by sirrobzeroone** [comment]
Thanks for the quick reply @sfan5, thats a little disappointing. I was going to underlay an image under hotbar so as to give it different look and not be constrained by the hard coded image size....kills that idea...

Hmm I guess i could look at the inventory hud api item...The only bit Im not sure about is how to get current selected hotbar item number either from inventory or somewhere else but Im sure were its stored or if its accessible in someway....i hope so :). 

 More work but I think worth it.

**Comment by sirrobzeroone** [comment]
I tried to work around the issue but the hud inventory type dosent have an alignment option, so im stuck. I added some comments too (https://github.com/minetest/minetest/issues/6296). 

Without some sort of fix or additional features looks like custom hotbars are off the menu :(.

**Comment by sirrobzeroone** [comment]
I part workaround this issue to get a custom hotbar running. I made a custom hotbar image then sliced the fixed sized center piece out of the middle leaving me with default "gui_hotbar.png" piece and a surrond piece which I then carefully aligned around the default hotbar using hud_add on joinplayer. 

Although of less use I also worked out a custom sized selector image although it experiences some latency when moving up/down along the hotbar fast but more than good enough for my purposes.  Similar a hud add image which updates to a new image which matches wield index...For my 10 slot hotbar this means 10 selector images 1 for each slot but again it works okay.

Basically smoke and mirrors and abit untidy but it works.
